versioned_docs/version-4.0.0-alpha.1/datastructures/storage-in-metadata.md

```diff
@@ -3,6 +3,212 @@ title: Metadata Format
 slug: /datastructures/storage-in-metadata
 ---
 
-:::caution
-TODO: Add documentation on the `storage ` field in the metadata.
-:::
+The storage layout of a contract is reflected inside the metadata. It allows third-party 
+tooling to work with contract storage and can also help to better understand the storage 
+layout of any given contract.
+
+Given a contract with the following storage:
+
+```rust
+#[ink(storage)]
+pub struct MyContract {
+    balance: Balance,
+    block: BlockNumber,
+    lazy: Lazy<bool>,
+}
+```
+
+The storage will be reflected inside the metadata as like follows:
+
+```json
+"root": {
+  "layout": {
+    "struct": {
+      "fields": [
+        {
+          "layout": {
+            "leaf": {
+              "key": "0x00000000",
+              "ty": 0
+            }
+          },
+          "name": "balance"
+        },
+        {
+          "layout": {
+            "leaf": {
+              "key": "0x00000000",
+              "ty": 1
+            }
+          },
+          "name": "block"
+        },
+        {
+          "layout": {
+            "root": {
+              "layout": {
+                "leaf": {
+                  "key": "0xb1f4904e",
+                  "ty": 2
+                }
+              },
+              "root_key": "0xb1f4904e"
+            }
+          },
+          "name": "lazy"
+        }
+      ],
+      "name": "MyContract"
+    }
+  },
+  "root_key": "0x00000000"
+}
+```
+
+We observe that the storage layout is represented as a tree, where tangible storage values 
+end up inside a `leaf`. Because of 
+[`Packed`](https://docs.rs/ink_storage_traits/4.0.0-beta.1/ink_storage_traits/trait.Packed.html) 
+encoding, leafs can share the same storage key, and 
+in order to reach them you'd need fetch and decode the whole storage cell under this key.
+
+A `root_key` is meant to either be used to directly access a `Packed` storage field 
+or to serve as the base key for calculating the actual keys needed to access values in 
+non-`Packed` fields (such as `Mapping`s).
+Layouts under a root key can contain `leaf`s directly, but also `struct`, `enum` or `array` 
+layout, which are used 
```

Comment:
The sentence cut off here is strange.

Reply:
This wasn't supposed to go in :see_no_evil: